### PR TITLE
CLOUDP-285943: update `atlas cluster get` to support Flex Clusters

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes
This PR updates the `atlas cluster get` command to support Flex Cluster. The command tries to call the cluster API first and, in the event of a 400 error, calls the flex cluster APIs 

Here is the error that you get when using Cluster API with flex clusters                                                                                 
`Error: https://cloud.mongodb.com/api/atlas/v2/groups/663268874ddd3b236fd2f107/clusters/Cluster22 GET: HTTP 400 Bad Request (Error code: "CANNOT_USE_FLEX_CLUSTER_IN_CLUSTER_API") Detail: Flex cluster Cluster22 cannot be used in the Cluster API. Reason: Bad Request. Params: [Cluster22]`

## Tests

<details>
  <summary>Dedicated Cluster </summary>

```bash
./bin/atlas cluster get Cluster0 -P prod                                                                                              
ID                         NAME       MDB VER   STATE
66d1b902e509e119a21ca4c8   Cluster0   7.0.15    IDLE
```

```json
./bin/atlas cluster get Cluster0 -P prod -o json                                                                                          
{
  "backupEnabled": false,
  "biConnector": {
    "enabled": false,
    "readPreference": "secondary"
  },
  "clusterType": "REPLICASET",
  "connectionStrings": {
    "standard": "mongodb://cluster0-shard-00-00.pkura.mongodb.net:27017,cluster0-shard-00-01.pkura.mongodb.net:27017,cluster0-shard-00-02.pkura.mongodb.net:27017/?ssl=true\u0026authSource=admin\u0026replicaSet=atlas-109val-shard-0",
    "standardSrv": "mongodb+srv://cluster0.pkura.mongodb.net"
  },
  "createDate": "2024-08-30T12:20:18Z",
  "diskSizeGB": 40,
  "diskWarmingMode": "FULLY_WARMED",
  "encryptionAtRestProvider": "NONE",
  "globalClusterSelfManagedSharding": false,
  "groupId": "663268874ddd3b236fd2f107",
  "id": "66d1b902e509e119a21ca4c8",
  "labels": [],
  "links": [
    {
      "href": "https://cloud.mongodb.com/api/atlas/v2/groups/663268874ddd3b236fd2f107/clusters/Cluster0",
      "rel": "self"
    },
    {
      "href": "https://cloud.mongodb.com/api/atlas/v2/groups/663268874ddd3b236fd2f107/clusters/Cluster0/backup/restoreJobs",
      "rel": "https://cloud.mongodb.com/restoreJobs"
    },
    {
      "href": "https://cloud.mongodb.com/api/atlas/v2/groups/663268874ddd3b236fd2f107/clusters/Cluster0/backup/snapshots",
      "rel": "https://cloud.mongodb.com/snapshots"
    }
  ],
  "mongoDBMajorVersion": "7.0",
  "mongoDBVersion": "7.0.15",
  "name": "Cluster0",
  "paused": true,
  "pitEnabled": false,
  "replicationSpecs": [
    {
      "id": "66d1b8fb08e4843269f73f1a",
      "numShards": 1,
      "regionConfigs": [
        {
          "electableSpecs": {
            "diskIOPS": 3000,
            "ebsVolumeType": "STANDARD",
            "instanceSize": "M30",
            "nodeCount": 3
          },
          "priority": 7,
          "providerName": "AWS",
          "regionName": "US_EAST_1",
          "analyticsAutoScaling": {
            "compute": {
              "enabled": true,
              "maxInstanceSize": "M40",
              "minInstanceSize": "M30",
              "scaleDownEnabled": true
            },
            "diskGB": {
              "enabled": true
            }
          },
          "analyticsSpecs": {
            "nodeCount": 0,
            "diskIOPS": 3000,
            "ebsVolumeType": "STANDARD",
            "instanceSize": "M30"
          },
          "autoScaling": {
            "compute": {
              "enabled": true,
              "maxInstanceSize": "M40",
              "minInstanceSize": "M30",
              "scaleDownEnabled": true
            },
            "diskGB": {
              "enabled": true
            }
          },
          "readOnlySpecs": {
            "nodeCount": 0,
            "diskIOPS": 3000,
            "ebsVolumeType": "STANDARD",
            "instanceSize": "M30"
          }
        }
      ],
      "zoneName": "Zone 1"
    }
  ],
  "rootCertType": "ISRGROOTX1",
  "stateName": "IDLE",
  "tags": [],
  "terminationProtectionEnabled": false,
  "versionReleaseSystem": "LTS"
}
```
</details>

<details>
  <summary>Flex Cluster </summary>

```bash
./bin/atlas cluster get Cluster22 -P prod                                                                                            
ID                         NAME        MDB VER   STATE
67406c8b74c55f6cea97513a   Cluster22   8.0.3     IDLE
```

```json
./bin/atlas cluster get Cluster22 -P prod -o json       
                                                                                  
{
  "backupSettings": {
    "enabled": true
  },
  "clusterType": "REPLICASET",
  "connectionStrings": {
    "standard": "mongodb://ac-41wbyjg-shard-00-00.xbvlwej.mongodb.net:27017,ac-41wbyjg-shard-00-01.xbvlwej.mongodb.net:27017,ac-41wbyjg-shard-00-02.xbvlwej.mongodb.net:27017/?ssl=true\u0026authSource=admin\u0026replicaSet=atlas-srjr8m-shard-0",
    "standardSrv": "mongodb+srv://cluster22.xbvlwej.mongodb.net"
  },
  "createDate": "2024-11-22T11:35:39Z",
  "groupId": "663268874ddd3b236fd2f107",
  "id": "67406c8b74c55f6cea97513a",
  "mongoDBVersion": "8.0.3",
  "name": "Cluster22",
  "providerSettings": {
    "backingProviderName": "AWS",
    "diskSizeGB": 5,
    "providerName": "FLEX",
    "regionName": "EU_WEST_1"
  },
  "stateName": "IDLE",
  "tags": [],
  "terminationProtectionEnabled": false,
  "versionReleaseSystem": "LTS"

```

### CODE REVIEW

File: internal/cli/clusters/describe.go [hunk 1]
```diff
@@ -49,12 +55,31 @@ var describeTemplate = `ID	NAME	MDB VER	STATE
...
+	return opts.Print(r)
+}
+
+func (opts *DescribeOpts) RunFlexCluster(err error) error {
+	apiError, ok := atlasClustersPinned.AsError(err)
+	if !ok {
 		return err
 	}
+
+	if *apiError.Error != cannotUseFlexWithClusterApisHTTPCode && *apiError.ErrorCode != cannotUseFlexWithClusterApisErrorCode {
```
Comment (jeroenvervaeke) [inline internal/cli/clusters/describe.go:70]: Afaik it's safe to only rely on the `ErrorCode`

File: internal/cli/clusters/describe.go [hunk 2]
```diff
@@ -49,12 +55,31 @@ var describeTemplate = `ID	NAME	MDB VER	STATE
...
+	r, err := opts.store.FlexCluster(opts.ConfigProjectID(), opts.name)
+	if err != nil {
+		return err
+	}
+
 	return opts.Print(r)
 }
 
-// atlas cluster(s) describe <clusterName> --projectId projectId.
+// DescribeBuilder atlas cluster(s) describe <clusterName> --projectId projectId.
```
Comment (jeroenvervaeke) [inline internal/cli/clusters/describe.go:82]: Is this change intentional?

File: internal/cli/clusters/describe.go
[the remark this answers is not in the document]
Comment (jeroenvervaeke) [reply internal/cli/clusters/describe.go:1]: I have no idea why gh made this thread with the title "general"

[the remark this answers is not in the document]
Comment (andreaangiolillo) [reply internal/cli/clusters/describe.go:1]: yes, I will add e2e tests once I have added all the CRUD operations

File: internal/cli/clusters/describe.go (hunk 1 above)
Comment (andreaangiolillo) [reply internal/cli/clusters/describe.go:70]: I will remove it
